server/api/products.js

```diff
@@ -48,58 +48,67 @@ router.get('/:productId', (req, res, next) => {
 
 //add new product
 router.post('/', function (req, res, next) {
-  console.log('request body in post is: ', req.body)
-  const { name, description, inventory, price } = req.body;
-  if (!name) {
-    return res.status(400).send('Invalid request; name required');
+  if (req.user && req.user.isAdmin) {
+    const { name, description, inventory, price } = req.body;
+    if (!name) {
+      return res.status(400).send('Invalid request; name required');
+    }
+    Product.create({ name, description, inventory, price })
+      .then(product => res.status(201).send(product))
+      .catch(next);
   }
-  Product.create({ name, description, inventory, price })
-    .then(product => res.status(201).send(product))
-    .catch(next);
+  return res.status(403).send('Invalid user credentials');
 });
 
 //update product for specific product id
 router.put('/:productId', (req, res, next) => {
-
-  const { name, description, inventory, price } = req.body;
-  const { productId } = req.params;
-  Product.update(
-    {
-      name, description, inventory, price
-    },
-    {
-      where: {
-        id: req.params.productId
+  console.log(req.user);
+  if (req.user && req.user.dataValues.isAdmin) {
+    const { name, description, inventory, price } = req.body;
+    const { productId } = req.params;
+    Product.update(
+      {
+        name, description, inventory, price
```

Comment:
following from my comment above - maybe better to do a `findOne` before update so that if any of these update fields are empty/falsey, we can say `name: name || product.name`

Reply:
The frontend component handles this.